playbooks/operationalizing/how-full-is-my-cluster.adoc

```diff
@@ -0,0 +1,130 @@
+To get an idea of the actual resources available in a node you can run (this command requires https://docs.openshift.com/container-platform/latest/install_config/cluster_metrics.html[cluster metrics] to be installed) +
+oc adm top node --heapster-namespace=openshift-infra --heapster-scheme=https <node-name> +
+Which will output something as follows: +
+NAME             CPU(cores)   CPU%      MEMORY(bytes)   MEMORY%   
+192.168.99.100   1007m        50%       5896Mi          49%
+
+This is the same node of the above example. Notice that the actual consumption of resources is much higher than what OpenShift estimated. The difference is due to the fact that in this node there are several pods that do not declare requests and limits.
+
+In summary, in order to fully being able to describe the available capacity of a cluster we should be able to answer two questions (and we need the answer them for at least memory and CPU):
+ . How much capacity OpenShift estimates is available based on the declared requests by the pods.
+ . How much capacity is really available based on current usage.
+
+How well the actual resource availability tracks the OpenShift estimated availability will depend on how well the pods have been sized and on the current load.  +
+Cluster administrators should watch the ratio between estimated resources and actual resources. They should also put in place policies to make sure that the two metrics stay as much as close as they can. This allows OpenShift to optimize allocation by increasing density, but at the same time guarantee the requested SLAs.
+
+== Monitoring the cluster available capacity
+
+Implementing a mature, enterprise-grade monitoring tool for OpenShift can take some time. I wanted to provide something that would allow to answer the resource availability question from day one. +
+One way is to script the above commands (oc describe node and oc adm top) and come up with some calculation to get to the answer. Another way is to use https://github.com/hjacobs/kube-ops-view[Kube-ops-view]. A ported version to support OpenShift is available https://github.com/raffaelespazzoli/kube-ops-view/tree/ocp[here]. +
+Kube-ops-view features a dashboard that allows to get information on the capacity of your cluster among other things. Here is an example of the dashboard:
+
+
+The nice thing of kube-ops-view is that you don’t have to install anything on the nodes and you can run it all on your laptop. You can also install in your cluster. +
+Kube-ops-view requires https://docs.openshift.com/container-platform/latest/install_config/cluster_metrics.html[metrics] to be installed and running correctly.
+
+=== Running kube-ops-view on you laptop
+
+For a local installation, you need to be logged in as a cluster administrator and then run the following: +
+oc proxy &
+docker run -it --net=host raffaelespazzoli/ocp-ops-view +
```

Comment:
Please put code in code block, for example:
```
oc proxy & docker run -it --net=host raffaelespazzoli/ocp-ops-view
```
may also want to make this more generalized (ie replace your name)

Reply:
it needs to be my name because that is where the image is